.eslintrc.js

```diff
@@ -0,0 +1,270 @@
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+    project: [
+      'packages/mgt/tsconfig.json',
+      'packages/mgt-element/tsconfig.json',
+      'packages/mgt-components/tsconfig.json',
+      'packages/mgt-react/tsconfig.json',
+      'packages/mgt-spfx/tsconfig.json',
+      'packages/mgt-spfx-utils/tsconfig.json',
+      'packages/providers/mgt-electron-provider/tsconfig.authenticator.json',
+      'packages/providers/mgt-electron-provider/tsconfig.provider.json',
+      'packages/providers/mgt-mock-provider/tsconfig.json',
+      'packages/providers/mgt-msal-provider/tsconfig.json',
+      'packages/providers/mgt-msal2-provider/tsconfig.json',
+      'packages/providers/mgt-proxy-provider/tsconfig.json',
+      'packages/providers/mgt-sharepoint-provider/tsconfig.json',
+      'packages/providers/mgt-teams-msal2-provider/tsconfig.json',
+      'packages/providers/mgt-teams-provider/tsconfig.json',
+      'packages/providers/mgt-teamsfx-provider/tsconfig.json'
+    ],
+    sourceType: 'module'
+  },
+  plugins: [
+    'eslint-plugin-jsdoc',
+    'eslint-plugin-prefer-arrow',
+    'eslint-plugin-react',
+    '@typescript-eslint',
+    '@typescript-eslint/tslint'
+  ],
+  root: true,
+  ignorePatterns: ['**/**-css.ts', '.eslintrc.js'],
+  rules: {
+    '@typescript-eslint/adjacent-overload-signatures': 'error',
+    '@typescript-eslint/array-type': [
+      'error',
+      {
+        default: 'array'
+      }
+    ],
+    '@typescript-eslint/ban-types': [
+      'error',
+      {
+        types: {
+          Object: {
+            message: 'Avoid using the `Object` type. Did you mean `object`?'
+          },
+          Function: {
+            message: 'Avoid using the `Function` type. Prefer a specific function type, like `() => void`.'
+          },
+          Boolean: {
+            message: 'Avoid using the `Boolean` type. Did you mean `boolean`?'
+          },
+          Number: {
+            message: 'Avoid using the `Number` type. Did you mean `number`?'
+          },
+          String: {
+            message: 'Avoid using the `String` type. Did you mean `string`?'
+          },
+          Symbol: {
+            message: 'Avoid using the `Symbol` type. Did you mean `symbol`?'
+          }
+        }
+      }
+    ],
+    '@typescript-eslint/consistent-type-assertions': 'error',
+    '@typescript-eslint/dot-notation': 'error',
+    '@typescript-eslint/explicit-function-return-type': 'off',
+    '@typescript-eslint/explicit-module-boundary-types': 'off',
+    '@typescript-eslint/indent': 'off',
+    '@typescript-eslint/member-delimiter-style': [
+      'off',
+      {
+        multiline: {
+          delimiter: 'none',
+          requireLast: true
+        },
+        singleline: {
+          delimiter: 'semi',
+          requireLast: false
+        }
+      }
+    ],
+    '@typescript-eslint/naming-convention': [
+      'warn',
+      {
+        selector: 'variable',
+        format: ['camelCase', 'UPPER_CASE'],
```

Comment:
```suggestion
        format: ['camelCase', 'UPPER_CASE', 'PascalCase'],
```
Adding `PascalCase` to silence naming convention errors in the generated react files.